jfix-armeria-aggregating-profiler/src/main/kotlin/ru/fix/armeria/aggregating/profiler/ProfiledError.kt

```diff
@@ -0,0 +1,93 @@
+package ru.fix.armeria.aggregating.profiler
+
+import com.linecorp.armeria.client.ResponseTimeoutException
+import com.linecorp.armeria.client.SessionProtocolNegotiationException
+import com.linecorp.armeria.client.endpoint.EndpointGroupException
+import com.linecorp.armeria.common.ClosedSessionException
+import com.linecorp.armeria.common.HttpStatus
+import com.linecorp.armeria.common.HttpStatusClass
+import com.linecorp.armeria.common.logging.RequestLog
+import com.linecorp.armeria.common.stream.ClosedStreamException
+import io.netty.handler.codec.http2.Http2Error
+import io.netty.handler.codec.http2.Http2Exception
+import ru.fix.armeria.commons.unwrapUnprocessedExceptionIfNecessary
+import java.net.ConnectException
+
+internal sealed class ProfiledError(
+    val latencyMetricRequired: Boolean = false,
+    typeMetricName: String,
+    vararg additionalMetrics: MetricTag
+) {
+
+    val metricTags: List<MetricTag> = listOf(
+        (MetricTags.ERROR_TYPE to typeMetricName),
+        *additionalMetrics
+    )
+
+    object ConnectRefused : ProfiledError(latencyMetricRequired = true, typeMetricName = "connect_refused")
+    object ConnectTimeout : ProfiledError(typeMetricName = "connect_timeout")
+    object NoAvailableEndpoint : ProfiledError(typeMetricName = "no_available_endpoint")
+    object RequestClosedSession
+        : ProfiledError(latencyMetricRequired = true, typeMetricName = "request_closed_session")
```

Comment:
maybe format code like this:
```suggestion
    object RequestClosedSession : ProfiledError(
        latencyMetricRequired = true,
        typeMetricName = "request_closed_session"
    )
```